File: demo/src/main/java/com/example/Agenda.java
package com.example;

public class Agenda {
    private Usuario[] registro;
    private int no_reg;

    public Agenda(int capacity) {
        registro = new Usuario[capacity];
        no_reg = 0;
    }

    public boolean agregar(Usuario u) {
        if (buscar((int)u.getId()) != -1 || no_reg >= registro.length) return false;
        registro[no_reg++] = u;
        return true;
    }

    public int buscar(int id) {
        for (int i = 0; i < no_reg; i++) {
            if (registro[i].getId() == id) return i;
        }
        return -1;
    }

    public boolean eliminar(int id) {
        int pos = buscar(id);
        if (pos == -1) return false;
        for (int i = pos; i < no_reg - 1; i++) {
            registro[i] = registro[i + 1];
        }
        registro[--no_reg] = null;
        return true;
    }

    public void mostrar() {
        for (int i = 0; i < no_reg; i++) {
            System.out.println("Nombre: " + registro[i].getNombre() + " | ID: " + registro[i].getId());
        }
    }

    public void mostrarUsuario(int id) {
        int pos = buscar(id);
        if (pos != -1)
            System.out.println(registro[pos]);
        else
            System.out.println("Usuario no encontrado.");
    }
}

File: python/Agenda.py
class Agenda:
    def __init__(self, capacidad):
        self.registro = [None] * capacidad
        self.no_reg = 0

    def buscar(self, id):
        for i in range(self.no_reg):
            if self.registro[i].get_id() == id:
                return i
        return -1

    def agregar(self, usuario):
        if self.buscar(usuario.get_id()) != -1 or self.no_reg >= len(self.registro):
            return False
        self.registro[self.no_reg] = usuario
        self.no_reg += 1
        return True

    def eliminar(self, id):
        pos = self.buscar(id)
        if pos == -1:
            return False
        for i in range(pos, self.no_reg - 1):
            self.registro[i] = self.registro[i + 1]
        self.registro[self.no_reg - 1] = None
        self.no_reg -= 1
        return True

    def mostrar(self):
        for i in range(self.no_reg):
            print(f"Nombre: {self.registro[i].get_nombre()} | ID: {self.registro[i].get_id()}")

    def mostrar_usuario(self, id):
        pos = self.buscar(id)
        if pos != -1:
            print(self.registro[pos])
        else:
            print("Usuario no encontrado.")
